### PR TITLE
EPMDJ-2171: Add Spinner while registering. Update error message

### PR DESCRIPTION
**Pull request by RomanAchilov**
Issue: EPMDJ-2171

### CODE REVIEW

File: src/pages/agent-registration-page/agent-registration-page-api/preregister-offline-agent.ts [hunk 1]
```diff
@@ -30,6 +31,8 @@ export function preregisterOfflineAgent(onSuccess?: () => void) {
       onSuccess && onSuccess();
     } catch ({ response: { data: { message } = {} } = {} }) {
       onError && onError(message || 'On-submit error. Server problem or operation could not be processed in real-time.');
+    } finally {
```
Comment (RomanDavlyatshin) [inline src/pages/agent-registration-page/agent-registration-page-api/preregister-offline-agent.ts:34]: Remove unnecessary use of `finally` statement? (unless you afraid that `onError` will throw)

File: src/pages/service-group-registration-page/service-group-registration-page.tsx [hunk 2]
```diff
@@ -58,7 +52,10 @@ export const ServiceGroupRegistrationPage = serviceGroupRegistrationPage(
         />
         <Wizard
           initialValues={serviceGroup}
-          onSubmit={handleRegisterAgent}
+          onSubmit={registerAgent(() => {
```
Comment (RomanDavlyatshin) [inline src/pages/service-group-registration-page/service-group-registration-page.tsx:55]: Do not pass callbacks, just move `registerAgent` contents right here.
You can move `axios` call to the separate function in the module scope to keep jsx readable.

File: src/components/wizard/wizard.tsx [hunk 3]
```diff
@@ -35,14 +35,15 @@ export const Wizard = wizard(({
 }: Props) => {
   const [{ currentStepIndex }, dispatch] = React.useReducer(wizardReducer, state);
   const [errorMessage, setErrorMessage] = React.useState('');
+  const [registering, setRegistrating] = React.useState(false);
```
Comment (RomanDavlyatshin) [inline src/components/wizard/wizard.tsx:38]: How about using a more generic name, e.g. `submitting` instead of `registering`?

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: src/components/wizard/wizard.module.scss [hunk 4]
```diff
@@ -15,4 +15,10 @@
   &__previous-button {
     margin-right: 16px;
   }
+
+  &__white-spinner {
+    & > div::after {
+      background-color: map-get($monochrome, white);
```
Comment (RomanDavlyatshin) [inline src/components/wizard/wizard.module.scss:21]: Btw, we could use utility classes for stuff like background color as well
@IliaSolovev what do you think? 😃

File: src/pages/agent-registration-page/agent-registration-page.tsx [hunk 5]
```diff
@@ -54,15 +55,11 @@ export const AgentRegistrationPage = agentRegistrationPage(
...
-            ? registerAgent(() => {
-              showMessage({ type: 'SUCCESS', text: 'Agent has been registered' });
-              push(`/full-page/${agentId}/${buildVersion}/dashboard`);
-            })
-            : preregisterOfflineAgent(() => {
-              showMessage({ type: 'SUCCESS', text: 'Offline agent has been preregistered' });
-              push('/agents');
-            })}
+          onSubmit={async (data: Agent) => {
+            agentId ? await registerAgent(data) : await preregisterOfflineAgent(data);
```
Comment (RomanDavlyatshin) [inline src/pages/agent-registration-page/agent-registration-page.tsx:59]: 😮  A ternary operator is used 3 times! 
IMO using `if..else` is better
```
if (agentId) {
  await registerAgent(data);
  showMessage(...)
  push(...)
} else {
  await preregisterOfflineAgent(data);
  showMessage(...)
  push(...)
}
```

File: src/pages/service-group-registration-page/service-group-registration-page.tsx [hunk 6]
```diff
@@ -58,7 +52,25 @@ export const ServiceGroupRegistrationPage = serviceGroupRegistrationPage(
...
-          onSubmit={handleRegisterAgent}
+          onSubmit={async (
+            {
+              id,
+              plugins,
+              name = '',
+              systemSettings,
+            }: Agent,
+          ) => {
+            await axios.patch(`/service-groups/${id}`, {
```
Comment (RomanDavlyatshin) [inline src/pages/service-group-registration-page/service-group-registration-page.tsx:63]: I suggest moving `axios.patch(...)` to the separate function `registerServiceGroup(data) {...}` in the module scope as well.
Just to be consistent and keep JSX clean.

File: src/pages/agent-registration-page/agent-registration-page.tsx [hunk 7]
```diff
@@ -128,3 +125,45 @@ export const AgentRegistrationPage = agentRegistrationPage(
...
+  name,
+  environment,
+  description,
+  plugins,
+  systemSettings,
+}: Agent) {
+  await axios.post('/agents', {
+    id,
+    name,
+    agentType: 'JAVA',
```
Comment (RomanDavlyatshin) [inline src/pages/agent-registration-page/agent-registration-page.tsx:140]: I believe `Test2Code.d.ts` file, provided by the backend team, includes agent type definitions.
If not, we could at least declare and use Enum here (in place of plain strings).
P.S. It's not a very hot issue right now, just a suggestion for the future.

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: src/components/wizard/wizard.module.scss (hunk 4 above)
Comment (RomanAchilov) [reply src/components/wizard/wizard.module.scss:21]: not in this case

**Pull request merged**